libc/utils/MPCWrapper/MPCUtils.cpp

```diff
@@ -0,0 +1,370 @@
+      : precision(precision),
+        mpc_rounding(MPC_RND(mpfr::get_mpfr_rounding_mode(rnd),
+                             mpfr::get_mpfr_rounding_mode(rnd))) {
+    mpc_init2(value, precision);
+    Complex<float> x_c = cpp::bit_cast<Complex<float>>(x);
+    mpfr_t real, imag;
+    mpfr_init2(real, precision);
+    mpfr_init2(imag, precision);
+    mpfr_set_flt(real, x_c.real, mpfr::get_mpfr_rounding_mode(rnd));
+    mpfr_set_flt(imag, x_c.imag, mpfr::get_mpfr_rounding_mode(rnd));
+    mpc_set_fr_fr(value, real, imag, mpc_rounding);
+    mpfr_clear(real);
+    mpfr_clear(imag);
+  }
+
+  template <typename XType,
+            cpp::enable_if_t<cpp::is_same_v<_Complex double, XType>, bool> = 0>
+  MPCNumber(XType x,
+            unsigned int precision = mpfr::ExtraPrecision<double>::VALUE,
+            RoundingMode rnd = RoundingMode::Nearest)
+      : precision(precision),
+        mpc_rounding(MPC_RND(mpfr::get_mpfr_rounding_mode(rnd),
+                             mpfr::get_mpfr_rounding_mode(rnd))) {
+    mpc_init2(value, precision);
+    Complex<double> x_c = cpp::bit_cast<Complex<double>>(x);
+    mpc_set_d_d(value, x_c.real, x_c.imag, mpc_rounding);
+  }
+
+  MPCNumber(const MPCNumber &other)
+      : precision(other.precision), mpc_rounding(other.mpc_rounding) {
+    mpc_init2(value, precision);
+    mpc_set(value, other.value, mpc_rounding);
+  }
+
+  MPCNumber &operator=(const MPCNumber &rhs) {
+    precision = rhs.precision;
+    mpc_rounding = rhs.mpc_rounding;
+    mpc_init2(value, precision);
+    mpc_set(value, rhs.value, mpc_rounding);
+    return *this;
+  }
+
+  MPCNumber(const mpc_t x, unsigned int p, mpc_rnd_t rnd)
+      : precision(p), mpc_rounding(rnd) {
+    mpc_init2(value, precision);
+    mpc_set(value, x, mpc_rounding);
+  }
+
+  ~MPCNumber() { mpc_clear(value); }
+
+  void getValue(mpc_t val) const { mpc_set(val, value, mpc_rounding); }
+
+  MPCNumber carg() const {
+    mpfr_t res;
+    mpc_t res_mpc;
+
+    mpfr_init2(res, precision);
+    mpc_init2(res_mpc, precision);
+
+    mpc_arg(res, value, MPC_RND_RE(mpc_rounding));
+    mpc_set_fr(res_mpc, res, mpc_rounding);
+
+    MPCNumber result(res_mpc, precision, mpc_rounding);
+
+    mpfr_clear(res);
+    mpc_clear(res_mpc);
+
+    return result;
+  }
+
+  MPCNumber cproj() const {
+
+    mpc_t res;
+
+    mpc_init2(res, precision);
+
+    mpc_proj(res, value, mpc_rounding);
+
+    MPCNumber result(res, precision, mpc_rounding);
+
+    mpc_clear(res);
+
+    return result;
+  }
+};
+
+namespace internal {
+
+template <typename InputType>
+cpp::enable_if_t<cpp::is_complex_v<InputType>, MPCNumber>
+unary_operation(Operation op, InputType input, unsigned int precision,
+                RoundingMode rounding) {
+  MPCNumber mpcInput(input, precision, rounding);
+  switch (op) {
+  case Operation::Carg:
+    return mpcInput.carg();
+  case Operation::Cproj:
+    return mpcInput.cproj();
+  default:
+    __builtin_unreachable();
```

Comment:
Above you had a switch with no default case and `__builtin_unreachable` outside the body of the `switch` statement. Pick one style; be consistent.